app/validators/questionnaire_schema.py

```diff
@@ -617,3 +656,26 @@ def get_section_id_for_block(self, block: Mapping) -> str | None:
     def get_section_id_for_block_id(self, block_id: str) -> str | None:
         if block := self.get_block(block_id):
             return self.get_section_id_for_block(block)
+
+    def get_list_collector_block_id_for_add_block(self, block_id) -> dict | None:
+        for block in self.blocks:
+            if (
+                block["type"] == "ListCollector"
+                and block["add_block"]["id"] == block_id
+            ):
+                return block["id"]
+
+    def get_list_collector_block_id_for_repeating_blocks(self, block_id) -> dict | None:
```

Comment:
I couldn't spot this being used, so removing it might be okay?

Reply:
Ah, unless this is for the other PR coming up?

Reply:
Again, it's not needed after these simplifications I showed you on Monday, not even needed for that next PR coming up.